ui/components/app/alert-system/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -162,7 +162,11 @@ export function MultipleAlertModal({
     initialAlertIndex === -1 ? 0 : initialAlertIndex,
   );
 
-  const selectedAlert = alerts[selectedIndex];
+  // If the selected alert is not found, default to the first alert
+  const selectedAlert = alerts[selectedIndex]
+    ? alerts[selectedIndex]
+    : alerts[0];
```

Comment:
[nit] we can shorten this to 
```suggestion
  const selectedAlert = alerts[selectedIndex] ?? alerts[0];
```